ocaml/xapi/quicktest_cbt.ml

```diff
@@ -47,61 +54,93 @@ let make_vdi_from ~session_id ~sR = (* SR has VDI.create as allowed *)
     ~sm_config:[]
 
 
-(**** Test declarations ****)
+(* ******************
+ * Test declarations
+ * ******************)
 
-(* Test enable/disable CBT, test cbt_enabled:false for new VDI *)
-let enable_disable_cbt_test ~session_id ~vDI =
-  let enable_cbt_test = make_test "Testing VDI.enable/disable_CBT" 4 in
+(* Test enable/disable_cbt, data_destroy, and snapshot update the necessary fields *)
+let vdi_data_destroy_test ~session_id ~vDI =
+  let data_destroy_test = make_test "Testing VDI.{enable/disable_cbt, data_destroy, snapshot}" 4 in
   try
-    start enable_cbt_test;
-    test_assert ~test:enable_cbt_test
-      (not (get_cbt_status ~session_id ~vDI))
-      ~msg:"VDI.cbt_enabled field should be set to false for new VDIs";
+    start data_destroy_test;
+    debug data_destroy_test "Enabling CBT on original VDI";
     VDI.enable_cbt ~session_id ~rpc:!rpc ~self:vDI;
-    test_assert ~test:enable_cbt_test
+    test_assert ~test:data_destroy_test
```

Comment:
To simplify things, we could specialize this for this test to avoid having to repeatedly pass the `~test` argument:
```ocaml
let data_destroy_test =  ... in
let test_assert = test_assert ~test:data_destroy_test in
...
```

Reply:
Maybe it'd be easier if I just renamed the test object to `test` in each individual test, so field punning takes care of the rest? Unless that loses clarity?

Reply:
I think it's the best if we minimize the amount of things that one has to read by completely omitting this parameter, in my opinion that makes the test clearer and easier to understand, but this is probably subjective.